### DISCUSSION TITLE
Reset Windows Update

### DISCUSSION THREAD

**Discussion by CalebBrendel** [ChrisTitusTech/winutil#2017] [Q&A] [answered]
Hello everyone! I am a HUGE fan of the winutil project. For some select customers, I can unable to pull the script due to domain restrictions. I have been fighting some Windows 10 update issues with a site and wanted to use the script and cannot. So instead, does anyone have documentation or know what command is used when clicking on "Reset Windows Update"?

If so, please let me know and thanks in advance! Screenshot below for reference.

![image](https://github.com/ChrisTitusTech/winutil/assets/109250665/b75f9e79-2229-4032-bd9f-8a0ec51c486f)

**Comment by og-mrk** [comment, marked as answer]
Hello @JumperTheHero 
This action under the Windows Config Tab isn't a simple one line command, but instead multiple registry edits (with some conditional checks), all have been put inside a PowerShell Function, which you can add to your PowerShell Profile (or you could download the script file, rename it if you want, and run the said script)

Here's a link to that script: https://github.com/ChrisTitusTech/winutil/blob/main/functions%2Fpublic%2FInvoke-WPFUpdatesdefault.ps1

**Reply by CalebBrendel** [reply]
@og-mrk Thank you so much for your reply! If you don't mind I have one other question. I am mainly dealing with a few customers that I work with for the MSP I work for. They are experiencing windows updates filling up the C drive and the updates aren't getting bombed out. So they essentially just keep filling up the drive. Is there something in Winutil that helps clear out those old updates and deletes them or a script you might know? Those are the two huge main windows issues I am juggling currently. Really appreciate the script for resetting windows! Thank you so much!

**Reply by og-mrk** [reply]
@JumperTheHero, I'm glad to help 😄
Please consider the previous reply of mine as an answer to this discussion's question, so people can find the answer easily 👍

Now, onto your question about a script or a command for cleaning up unused windows update (update that won't affect your system of you remove them, as they are already applied in  lets say, the latest cumulative update).
WinUtil, _at the time of writing_, doesn't support an action for cleaning up Windows update, but I do know of a script that's known among Windows power users, which's Sophia Script, there's a lot of features in that tool, but I think the one you're looking for is called `Windows Cleanup` and `Windows Cleanup Notification`, which's documented in the `README` of the repo, here's a link to Sophia Script Repo: https://github.com/farag2/Sophia-Script-for-Windows

**Reply by CalebBrendel** [reply]
@og-mrk Thanks again man! Really appreciate it! Take it easy! I will look into the Sophia Script! Thanks a bunch again! :)
